src/renderer/services/file-management-system/index.ts

```diff
@@ -48,8 +48,7 @@ export default class FileManagementSystem {
   private readonly jss: JobStatusService;
   private readonly mms: MetadataManagementService;
 
-  private static readonly CHUNKS_INFLIGHT_REQUEST_MEMORY_USAGE_MAX = 40 * 1024 * 1024; // 40 mb
-  private static readonly CHUNKS_CEILING_INFLIGHT_REQUEST_CEILING = 10; //ceiling on concurrent chunk requests (even if more can fit in memory)
+  private static readonly CHUNKS_CEILING_INFLIGHT_REQUEST_CEILING = 20; //ceiling on concurrent chunk requests (even if more can fit in memory)
```

Comment:
To assess the `chunksInFlight` limit, I did some 10GB uploads with the same code as in this PR, but with `hasher.update()` commented out (e.g. upload ID 1cbca2d137d042c882797397753e9bde). In this test:
* `chunksInFlight` was consistently around 18-20
* The chunk size was just 50MB, not the 500MB chunks of a big 5TB upload, so the heap usage I saw is maybe 10x smaller than what we might see there?
* Maximum heap usage was just 80MB. This is surprisingly low, since I was seeing heap usage at 670MB with the MD5 computation turned on
My interpretation of this is that when the application is not CPU-bound it is able to use memory more efficiently (e.g., it can finish processing some data and then deallocate it sooner). This makes me think that changes to FUA which let us get to 20 chunks in flight are likely to also change the way the app uses memory, so we can cross that bridge when we get to it.

In practice I think it will be hard for this limit to matter until we do some MD5 performance work.

(While I was at it, I tried bumping the limit up to 80, and it easily reached 80 and was faster.)

Reply:
Update: I tried another test to attempt to cause more heap usage.
I applied the following patch to remove the limit on chunks in flight and only skip over the CryptoJS library's `this.hasher.update(word)` keeping our heap-based computation of the WordArray:
```
diff --git a/src/renderer/services/file-management-system/Md5Hasher.ts b/src/renderer/services/file-management-system/Md5Hasher.ts
index 92fe5f7f..d3d223c8 100644
--- a/src/renderer/services/file-management-system/Md5Hasher.ts
+++ b/src/renderer/services/file-management-system/Md5Hasher.ts
@@ -79,7 +79,7 @@ export default class Md5Hasher{
             wa[(i / 4) | 0] |= chunk[i] << (24 - 8 * i);
         }
         const word = CryptoJS.lib.WordArray.create(wa, chunk.length);
-        this.hasher.update(word);
+        // this.hasher.update(word);
     }
 
     /**
diff --git a/src/renderer/services/file-management-system/index.ts b/src/renderer/services/file-management-system/index.ts
index 94483728..0c453076 100644
--- a/src/renderer/services/file-management-system/index.ts
+++ b/src/renderer/services/file-management-system/index.ts
@@ -48,7 +48,7 @@ export default class FileManagementSystem {
   private readonly jss: JobStatusService;
   private readonly mms: MetadataManagementService;
 
-  private static readonly CHUNKS_CEILING_INFLIGHT_REQUEST_CEILING = 20; //ceiling on concurrent chunk requests (even if more can fit in memory)
+  private static readonly CHUNKS_CEILING_INFLIGHT_REQUEST_CEILING = 2000000; //ceiling on concurrent chunk requests (even if more can fit in memory)
 
   /**
    * Returns JSS friendly UUID to group files
 ```
 This showed lower concurrency (as expected; see screenshot) and higher memory usage (as expected) at 457MB. However, that's still lower than the 670MB heap usage with this PR's code.